examples/graphene_apps/python_worker/avalon_worker/work_order_processor.py

```diff
@@ -0,0 +1,364 @@
+import os
+import logging
+import socket
+import base64
+import random
+import secrets
+import string
+import json
+from hashlib import sha256
+import avalon_worker.receive_request as receive_request
+import avalon_worker.worker_encryption as worker_encryption
+import avalon_worker.worker_signing as worker_signing
+import avalon_worker.worker_hash as worker_hash
+import avalon_worker.workload_processor as workload_processor
+from avalon_worker.error_code import WorkerError
+import avalon_worker.utility.jrpc_utility as jrpc_utility
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+# -------------------------------------------------------------------------
+
+
+def main(args=None):
+    """
+    Graphene worker main function
+    """
+    # Create Process work order object
+    wo_processor = WorkOrderProcessor()
+    # Setup ZMQ channel to receive work order
+    url = "tcp://*:7777"
+    zmq_socket = receive_request.ZmqSocket(url, wo_processor)
+    logger.info("Start listening to ZMQ for work orders")
+    # Start listener and wait for new request
+    zmq_socket.start_zmq_listener()
+
+# -------------------------------------------------------------------------
+
+
+class WorkOrderProcessor():
+    """
+    Graphene work order processing class.
+    """
+
+# -------------------------------------------------------------------------
+
+    def __init__(self):
+        """
+        Constructor to generate worker signing and encryption keys.
+        """
+        self.generate_worker_keys()
+
+# -------------------------------------------------------------------------
+
+    def generate_worker_keys(self):
+        """
+        Generates worker signing and encryption keys.
+        """
+        # Generate worker signing key
+        logger.info("Generate worker signing and encryption keys")
+        self.sign = worker_signing.WorkerSign()
+        self.sign.generate_signing_key()
+        self.worker_public_sign_key = self.sign.get_public_sign_key()
+        # Generate worker encryption key
+        self.encrypt = worker_encryption.WorkerEncrypt()
+        self.encrypt.generate_rsa_key()
+        self.worker_public_enc_key = self.encrypt.get_rsa_public_key()
+        # Sign worker encryption key hash
+        hash_obj = worker_hash.WorkerHash()
+        hash_val = hash_obj.compute_message_hash(self.worker_public_enc_key)
+        self.worker_public_enc_key_sign = self.sign.sign_message(hash_val)
+
+# -------------------------------------------------------------------------
+
+    def process_work_order(self, msg):
+        """
+        Process Avalon work order or signup request.
+
+        Parameters :
+            msg: JSON formatted string containing the request to execute.
+        Returns :
+            JSON RPC response containing result or error.
+        """
+        try:
+            json_msg = json.loads(msg)
+        except json.decoder.JSONDecodeError:
```

Comment:
Is it being ascertained that the msg would not be None? The exception would be a different one in that case - TypeError. We could see other exceptions if that's not caught.

Reply:
Yes. Better to catch other exception with `Exception`